File: backend/models/benchmark.py
# Benchmark Pydantic Models

from pydantic import BaseModel, Field
from enum import Enum


class BenchmarkRiskLevel(str, Enum):
    NORMAL = "normal"
    ELEVATED = "elevated"
    EXTREME = "extreme"


class BenchmarkResultResponse(BaseModel):
    id: str = Field(alias="_id")
    line_item_id: str
    bill_id: str
    code: str
    benchmark_source: str
    medicare_rate: float
    typical_low: float
    typical_median: float
    typical_high: float
    billed_amount: float
    deviation_percentage: float
    deviation_score: int
    risk_level: BenchmarkRiskLevel

    class Config:
        populate_by_name = True


class BenchmarkSummary(BaseModel):
    items_above_typical: int
    estimated_savings_low: float
    estimated_savings_high: float

File: src/app/types/benchmark.ts
// Benchmark types — BenchmarkResult, DeviationScore, BenchmarkRiskLevel

export type BenchmarkRiskLevel = "normal" | "elevated" | "extreme";

export interface BenchmarkCallResolution {
  status: "resolved" | "adjusted" | "denied";
  previous_amount: number;
  new_amount: number;
  note: string;
  call_id: string;
}

export interface BenchmarkResult {
  _id: string;
  line_item_id: string;
  bill_id: string;
  code: string;
  benchmark_source: string;
  medicare_rate: number;
  typical_low: number;
  typical_median: number;
  typical_high: number;
  billed_amount: number;
  deviation_percentage: number;
  deviation_score: number;
  risk_level: BenchmarkRiskLevel;
  call_resolution?: BenchmarkCallResolution;
}

export interface BenchmarkSummary {
  items_above_typical: number;
  estimated_savings_low: number;
  estimated_savings_high: number;
}
